### ISSUE TITLE
Error during tutorial

### ISSUE THREAD

**Issue by tvanlaar** [biofuture/Ublastx_stageone#11]
Hi, I am trying to run through the tutorial and I keep getting the error that it cannot execute the binary file.  I have pasted the entire string here starting with the command I entered, followed by all the errors.  I appreciate any help!

swuser-129-8-216-187:example tvanlaar$ ../ublastx_stage_one_version2 -i inputfqs/ -m meta-data.txt -o testout -n 5
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 4: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/usearch: cannot execute binary file
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 5: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/usearch: cannot execute binary file
No such file or directory  at /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/extract_usearch_reads.pl line 8.
No such file or directory  at /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/extract_usearch_reads.pl line 8.
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 9: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/usearch: cannot execute binary file
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 10: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/usearch: cannot execute binary file
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 14: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/diamond: cannot execute binary file
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 15: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/diamond: cannot execute binary file
cat: testout/STAS.uscmg_1.dmd: No such file or directory
cat: testout/STAS.uscmg_2.dmd: No such file or directory
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 20: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/usearch: cannot execute binary file
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 21: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/usearch: cannot execute binary file
No such file or directory  at /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/extract_usearch_reads.pl line 8.
No such file or directory  at /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/extract_usearch_reads.pl line 8.
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 25: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/usearch: cannot execute binary file
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 26: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/usearch: cannot execute binary file
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 30: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/diamond: cannot execute binary file
testout/ublastx_bash_Tue-Mar-6-16:26:39-2018.sh: line 31: /Users/tvanlaar/Desktop/ARGsOAP/Ublastx_stageone/bin/diamond: cannot execute binary file
cat: testout/SWHAS104.uscmg_1.dmd: No such file or directory
cat: testout/SWHAS104.uscmg_2.dmd: No such file or directory
wrong run

**Comment by biofuture** [comment]
chmod 755 usearch

**Comment by tvanlaar** [comment]
Thank you for your help.  Now when I run the test data, it looks like this where the final extracted.fa file has no data

../ublastx_stage_one -i inputfqs/ -m meta-data.txt -o testout -n 5
00:00 29Mb    100.0% Reading rows
00:00 29Mb   Reading pointers...done.
00:00 29Mb   Reading db seqs...done.

WARNING: Max OMP threads 4

00:12 50Mb    100.0% Searching STAS_1.fa, 0.1% matched
00:00 29Mb    100.0% Reading rows
00:00 29Mb   Reading pointers...done.
00:00 29Mb   Reading db seqs...done.

WARNING: Max OMP threads 4

00:09 50Mb    100.0% Searching STAS_2.fa, 0.0% matched
00:00 29Mb    100.0% Reading rows
00:00 29Mb   Reading pointers...done.
00:00 29Mb   Reading db seqs...done.

WARNING: Max OMP threads 4

00:20 52Mb    100.0% Searching STAS_1.fa, 0.1% matched
00:00 29Mb    100.0% Reading rows
00:00 29Mb   Reading pointers...done.
00:00 29Mb   Reading db seqs...done.

WARNING: Max OMP threads 4

00:20 52Mb    100.0% Searching STAS_2.fa, 0.1% matched
00:00 29Mb    100.0% Reading rows
00:00 29Mb   Reading pointers...done.
00:00 29Mb   Reading db seqs...done.

WARNING: Max OMP threads 4

00:11 50Mb    100.0% Searching SWHAS104_1.fa, 0.1% matched
00:00 29Mb    100.0% Reading rows
00:00 29Mb   Reading pointers...done.
00:00 29Mb   Reading db seqs...done.

WARNING: Max OMP threads 4

00:12 50Mb    100.0% Searching SWHAS104_2.fa, 0.1% matched
00:00 29Mb    100.0% Reading rows
00:00 29Mb   Reading pointers...done.
00:00 29Mb   Reading db seqs...done.

WARNING: Max OMP threads 4

00:21 52Mb    100.0% Searching SWHAS104_1.fa, 0.1% matched
00:00 29Mb    100.0% Reading rows
00:00 29Mb   Reading pointers...done.
00:00 29Mb   Reading db seqs...done.

WARNING: Max OMP threads 4

00:19 52Mb    100.0% Searching SWHAS104_2.fa, 0.1% matched
Wrong ID gi|926400371|ref|WP_053728483.1|

**Comment by tvanlaar** [comment]
Ok that worked except now I'm getting one last error: 
Wrong ID gi|740643397|ref|WP_038428846.1|
The extracted.fa file is empty, but the other .fa files in testout folder DO have data

**Comment by biofuture** [comment]
Do you change the database of the 30 KOs? This gene "
gi|740643397|ref|WP_038428846.1| " should not exist in the database.

Regards,

Xiao-Tao Jiang, Ph.D.
Department of Civil Engineering
The University of HongKong

2018-03-09 1:49 GMT+08:00 tvanlaar <notifications@github.com>:

**Comment by tvanlaar** [comment]
I went ahead and reinstalled and everything worked great.  Thanks so much!

**Comment by anotherXinzhu** [comment]
>           chmod 755 usearch

Hi Xiao-Tao,

I have the same issue and chmod 755 usearch doesn't help. 
At first I was running on Ubuntu, I tried a lot of times and could't figure out a way to change permission of the "usearch" file. 
Then I switch to mac OS and yes I was able to change perssion (but the permission was already -rwxr-xr-x when I first cloned the repo). It still shows all these same errors end with ": cannot execute binary file" and "no such file or directory at ......extract_usearch_reads.pl line 8." extract.fa is empty and there is only a head line in meta_data_online.txt.

**Comment by biofuture** [comment]
Probably you just need to reinstall it!  check whether all the files under bin are executable, with chmod -R 755 bin/

**Comment by anotherXinzhu** [comment]
>           Probably you just need to reinstall it!  check whether all the files under bin are executable, with chmod -R 755 bin/

Hi Xiaotao, 

Thank you so much for your reply.  
Finally got it work but here is what I went through ( think it might be good to share):

Firstly I was using ubuntu and the usearch doesn't work. I checked on the usearch website and they have three versions of usearch for mac OS, linux, and windows repectively. I downloaded the one for windows, put it in the bin folder,  replace the "usearch" to "usearch.exe" in the codes in argoap_pipeline_stageone_version2. Yes the original problem was solved but I got another problem: the database files ".udb" could not be opened. It is too complicated for me to make further changes so I gave up...

Then I run on mac OS, and did the same thing: downloading the mac version of usearch and putting it into the bin folder. It worked! But I had similar problem with diamond....

Then my colleague showed up and taught me to use the server with linux system. Problem solved and you are right, it can only be run with linux system... LOL
